bids-validator/validators/bids/checkConsistency.js

```diff
@@ -0,0 +1,228 @@
+
+  if (jsonUnit === 'um') {
+    if (omeUnit === 'mm') {
+      return 1000
+    } else if (omeUnit === 'nm') {
+      return 0.001
+    }
+  } else if (jsonUnit === 'mm') {
+    if (omeUnit === 'µm') {
+      return 0.001
+    } else if (omeUnit === 'nm') {
+      return 0.000001
+    }
+  } else if (jsonUnit === 'nm') {
+    if (omeUnit === 'mm') {
+      return 1000000
+    } else if (omeUnit === 'µm') {
+      return 1000
+    }
+  }
+}
+
+const getMatchingComponents = (omeFiles, jsonFiles) => {
+  let components = []
+  omeFiles.forEach(omeFile => {
+    let possibleJsonPath = omeFile.relativePath
+      .replace('.tif', '')
+      .replace('.ome', '.json')
+
+    let potentialSidecars = utils.files.potentialLocations(possibleJsonPath)
+
+    // Find the path before filename
+    let regExp = new RegExp('(.*/).*')
+    let preFilename = regExp.exec(possibleJsonPath)[1]
+
+    // Retrieve all json files at the same path with ome-tiff files
+    let jsonPaths = potentialSidecars.filter(path => {
+      let jsonpath = regExp.exec(path)[1]
+      return jsonpath === preFilename
+    })
+
+    // if possible json paths are not empty
+    jsonFiles.forEach(jsonFile => {
+      // if possible json file exists
+      if (jsonPaths.includes(jsonFile.relativePath)) {
+        components.push({
+          omeFile: omeFile,
+          jsonFilePath: jsonFile.relativePath,
+        })
+      }
+    })
+  })
+
+  return components
+}
+
+const checkConsistency = (omeFiles, jsonFiles, jsonContentsDict) => {
+  let issues = []
+
+  let components = getMatchingComponents(omeFiles, jsonFiles)
+
+  // if at least one ome-tiff file has no corresponding json file
+  if (components.length !== omeFiles.length) {
+    issues.push(new Issue({ code: 223 }))
+  }
+
+  const omeIssuesPromises = components.map(component => {
+    return utils.limit(
+      () =>
+        new Promise(async resolve => {
+          let jsonData = jsonContentsDict[component.jsonFilePath]
+          let omeData = await getOMETiffData(component.omeFile)
+          let optionalFieldsIssues = await validateOptionalFields(
+            omeData,
+            jsonData,
+          )
+          let pixelSizeIssues = await validatePixelSize(omeData, jsonData)
+          let matrixIssues = validateChunkTransformationMatrix(
+            component.jsonFilePath,
+            component.omeFile.path,
+            jsonData,
+          )
+          issues = issues
+            .concat(optionalFieldsIssues)
+            .concat(pixelSizeIssues)
+            .concat(matrixIssues)
+          return resolve()
+        }),
+    )
+  })
+
+  return new Promise(resolve =>
+    Promise.all(omeIssuesPromises).then(() => resolve(issues)),
+  )
+}
+
+const validateOptionalFields = async (omeData, jsonData) => {
+  let issues = []
+
+  let fields = {
+    Immersion: 'Immersion',
+    NumericalAperture: 'LensNA',
+    NominalMagnification: 'Magnification',
```

Comment:
```suggestion
    Immersion: 'Immersion',
    NumericalAperture: 'LensNA',
    Magnification: 'NominalMagnification',
```
The last one is inverted, assuming the order is `JSON field: 'OMETIFF field'`

Reply:
Done